build-advisories/src/Roave/SecurityAdvisories/VersionConstraint.php

```diff
@@ -0,0 +1,207 @@
+<?php
+
+namespace Roave\SecurityAdvisories;
+
+/**
+ * A simple version constraint - naively assumes that it is only about ranges like ">=1.2.3,<4.5.6"
+ */
+final class VersionConstraint
+{
+    const CLOSED_RANGE_MATCHER    = '^>(=?)\s*((\d+.)*\d+)\s*,\s*<(=?)\s*((\d+.)*\d+)$';
```

Comment:
yeah, this is because regexper expect Javascript regexes, and JS does not support possessive quantifiers.